pkg/templating/template.go

```diff
@@ -0,0 +1,114 @@
+package templating
+
+import (
+	"fmt"
+	"html/template"
+	"strings"
+
+	"github.com/Masterminds/sprig/v3"
+)
+
+// The split functions cannot be replaced with a single generic one, because there is
+// no common type between maps and slices
+
+// Given a map `src`, return two maps where the first contains key-value pairs that are not
+// templates, and the latter containers only key-value pairs that are templates.
+func SplitTemplatedMapItems(src map[string]string) (map[string]string, map[string]string) {
+	nonTemplateKeyPairs := make(map[string]string, len(src))
+	templateKeyPairs := make(map[string]string, len(src))
+
+	for key, value := range src {
+		if IsStringATemplate(value) {
+			templateKeyPairs[key] = value
+		} else {
+			nonTemplateKeyPairs[key] = value
+		}
+	}
+
+	return nonTemplateKeyPairs, templateKeyPairs
+}
+
+// Given a slice `src`, return two slices where the first contains key-value pairs that are not
+// templates, and the latter containers only key-value pairs that are templates. Values of the
+// opposing types will be set to an empty string, to preserve the position of all items.
+func SplitTemplatedListItems(src []string) ([]string, []string) {
+	nonTemplateKeyPairs := make([]string, len(src))
```

Comment:
The name of the variable should be `nonTemplateList` or something else?
And the size of the slice can be `0, len(src)`? This might be more efficient? I don't know for sure though.

Reply:
> The name of the variable should be `nonTemplateList` or something else?

Yep, you're right. Copy/paste error. Made a similar mistake below as well.

> And the size of the slice can be `0, len(src)`? This might be more efficient? I don't know for sure though.

Normally yes but in this specific case no. The second argument is the length, and the last is the capacity. The goal here is to produce two arrays of the same _length_, with nil/default/zero values for unwanted items. This preserves the order of items in the list so that they can later be combined together again, after templating.